### ISSUE TITLE
Twitter login with preregistered app

### ISSUE THREAD

**Issue by MartiBarb** [strohne/Facepager#181]
Hi everyone! 
I've registered my own App in my Twitter Developer Account. I have the API Key, API Key Secret, Access Token and Access Token Secret. I have inserted them in the corresponding fields in Facepager, but when I try to login to Twitter, it asks me to return to my project and it provides a PIN that should be inserted to complete the authorisation. Where do I have to insert such PIN?

Thank you for your help!

Martina

**Comment by sblanar** [comment]
I have the same problem.
Did you ever resolved it?

**Comment by MartiBarb** [comment]
No, I didn't. To do my research, I left Facepager to use twarc2 (https://twarc-project.readthedocs.io/en/latest/twarc2_en_us/). Maybe this will help you!

**Comment by strohne** [comment]
Hi there. sorry for the late response - may I ask you to ask such questions regarding how to use Facepager in the Facepager User group on Facebook? We usually only proceed development issues here.

With regard to your problem: I guess the authorization settings of you App do not match what Facepager expects. When using Twitter API v2, the easiest way is to use the OAuth Client Credential procedure (select it in the settings) with the login URL set to https://api.twitter.com/oauth2/token/ . Make sure to select the v2 basepath in the query setup of Facepager.

You only need the API key (->Client ID field) and the API secret key (-> Client Secret Field). Clicking the login button fetches an access token.

Anytime, the Twitter module is too restrictive, you can use the Generic module as the builtin presets for Twitter v2 do.
